<launch>

  <!-- =================================================================================
                        Launch Task Management Stuff
       ================================================================================= -->

  <arg name="smach_viewer" doc="(bool) bring up the smach viewer"/>
  <arg name="bringup_arm1" doc="(bool) launch arm1"/>

  <!-- node for moving ring to destination peg -->
  <node name="move_ring_motion"
    pkg="dmp_action"
    type="dmp_motion_server.py" />

  <!-- node for reaching towards ring -->
  <node name="reach_ring_motion"
    pkg="dmp_action"
    type="dmp_motion_server.py" />

  <!-- wait for the ring to be in range with this action -->
  <node name="wait_for_ring"
    pkg="peg_assist_demo"
    type="WaitForRing.py" />

  <!-- startup script to make sure the arms are running the right controllers -->
  <node name="peg_demo_startup"
    pkg="peg_assist_demo"
    type="startup.py" />

  <!-- node for running the plan -->
  <node name="execution_task" pkg="peg_assist_demo" type="task.py" output="screen">
    <param name="move_ring_file" value="($find peg_assist_demo)/data/move_ring.yaml"/>
    <param name="grab_ring_file" value="($find peg_assist_demo)/data/grab_ring.yaml"/>
  </node>

  <!-- service node for controlling the robot arm, interfaces with MoveIt and orocos -->
  <node name="Arm2Commander" pkg="peg_assist_demo" type="ArmCommander.py">
    <param name="hand_topic" value="/gazebo/w2barrett_manager/hand/cmd"/>
  </node>

  <!-- view execution progress -->
  <node if="$(arg smach_viewer)"
    name="smach_viewer"
    pkg="smach_viewer"
    type="smach_viewer.py" />

  <!-- publish planning information -->
  <node name="planning_scene_script" pkg="peg_assist_demo" type="publish_planning_scene.py" output="screen">
    <param name="wait_for" value="/gazebo/w2barrett_manager/move_group"/>
    <param name="service" value="/gazebo/wam2/publish_planning_scene"/>
    <param name="wait_time" value="0.0"/>
    <param name="rate" value="0.1"/>
  </node>

  <node name="planning_scene_script_arm1" pkg="peg_assist_demo" type="publish_planning_scene.py" output="screen"
    if="$(arg bringup_arm1)">
    <param name="wait_for" value="/gazebo/barrett_manager/move_group"/>
    <param name="service" value="/gazebo/wam/publish_planning_scene"/>
    <param name="wait_time" value="0.0"/>
    <param name="rate" value="0.1"/>
  </node>
</launch>
